### ISSUE TITLE
Setting style attribute on table makes messed up value

### ISSUE THREAD

**Issue by lean-jeremy** [ckeditor/ckeditor5#15269]
## 📝 Provide detailed reproduction steps (if any)

I use setAttribute to set both the class and style of a table.
```js
editor.model.change ( writer => {
      writer.setAttribute("class", "test", table)
      writer.setAttribute("style", "background-color:Tomato;", table)
});
```

### ✔️ Expected result

```html
<figure class="table test" style="background-color:Tomato;"> 
   <table>
...
```

### ❌ Actual result
The actual result has this weird string for the style value. I can see some similarity/pattern ("2:c;3:k;4:g;5:r;6:o;7:u;8:n;9:d;" = ckground?)
```html
<figure class="table test" style="0:b;10:-;11:c;12:o;13:l;14:o;15:r;16::;17:T;18:o;19:m;1:a;20:a;21:t;22:o;23:;;2:c;3:k;4:g;5:r;6:o;7:u;8:n;9:d;"> 
   <table>
...
```

### ❓ Possible solution

Perhaps there is something wrong with the style conversion?
```js
editor.conversion.attributeToAttribute({
    model: "style",
    view: modelAttributeValue => {
        key: "style"
        value: modelAttributeValue
    }
})
```

## 📃 Other details

* Browser: Chrome
* OS: Windows
* First affected CKEditor version: 40.0.0
* Installed CKEditor plugins: 
Alignment,
Autoformat,
Base64UploadAdapter,
BlockQuote,
Bold,
CloudServices,
Essentials,
FindAndReplace,
FontBackgroundColor,
FontColor,
FontFamily,
FontSize,
GeneralHtmlSupport,
Heading,
HorizontalLine,
Image,
ImageCaption,
ImageStyle,
ImageToolbar,
ImageUpload,
Indent,
Italic,
Link,
List,
ListProperties,
MediaEmbed,
PageBreak,
Paragraph,
PasteFromOffice,
RemoveFormat,
SelectAll,
SourceEditing,
SpecialCharacters,
SpecialCharactersArrows,
SpecialCharactersCurrency,
SpecialCharactersEssentials,
SpecialCharactersLatin,
SpecialCharactersMathematical,
SpecialCharactersText,
Strikethrough,
Style,
Subscript,
Superscript,
Table,
TableCaption,
TableCellProperties,
TableColumnResize,
TableProperties,
TableToolbar,
TextTransformation,
Underline

I appreciate any help 🙏

**Comment by Reinmar** [comment]
`setAttribute()` certainly shouldn't behave this way.

But the actual method to set a style is this one: https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_view_downcastwriter-DowncastWriter.html#function-setStyle:OBJECT.

**Comment by Reinmar** [comment]
Disregard my previous comment. I didn't notice this is a model change.

I think I saw a report regarding trying to keep the entire style of an element in the model as a style attribute. It should certainly be convertible.

**Comment by lean-jeremy** [comment]
@Reinmar @Witoso 
Thank you very much for the help 🙏, I've had luck using Reinmar's workaround [here].(https://github.com/ckeditor/ckeditor5/issues/4517#issuecomment-496133525)

Now the style is converted properly. This fits my goals, so this ticket can be closed but I'll leave that up to you since [#4517](https://github.com/ckeditor/ckeditor5/issues/4517) seems to be ongoing.

**Comment by Witoso** [comment]
Glad it helped, please +1 the https://github.com/ckeditor/ckeditor5/issues/4517, and I will close this one :)
